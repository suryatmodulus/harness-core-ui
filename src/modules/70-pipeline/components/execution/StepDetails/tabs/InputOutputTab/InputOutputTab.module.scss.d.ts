/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly ioTab: string
  readonly ioRow: string
  readonly header: string
  readonly key: string
  readonly value: string
  readonly valueText: string
  readonly panel: string
  readonly panelTitle: string
}
export default styles
