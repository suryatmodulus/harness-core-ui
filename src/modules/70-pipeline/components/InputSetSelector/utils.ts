import type { IconName, SelectOption } from '@harness/uicore'
import type { EntityGitDetails, InputSetSummaryResponse } from 'services/pipeline-ng'
import css from './InputSetSelector.module.scss'

type InputSetLocal = InputSetSummaryResponse & SelectOption
export interface InputSetValue extends InputSetLocal {
  type: InputSetSummaryResponse['inputSetType']
  gitDetails?: EntityGitDetails
}

export const getIconByType = (type: InputSetSummaryResponse['inputSetType']): IconName => {
  return type === 'OVERLAY_INPUT_SET' ? 'step-group' : 'yaml-builder-input-sets'
}

export const onDragStart = (event: React.DragEvent<HTMLLIElement>, row: InputSetValue): void => {
  event.dataTransfer.setData('data', JSON.stringify(row))
  event.currentTarget.classList.add(css.dragging)
}

export const onDragEnd = (event: React.DragEvent<HTMLLIElement>): void => {
  event.currentTarget.classList.remove(css.dragging)
}

export const onDragLeave = (event: React.DragEvent<HTMLLIElement>): void => {
  event.currentTarget.classList.remove(css.dragOver)
}

export const onDragOver = (event: React.DragEvent<HTMLLIElement>): void => {
  if (event.preventDefault) {
    event.preventDefault()
  }
  event.currentTarget.classList.add(css.dragOver)
  event.dataTransfer.dropEffect = 'move'
}
