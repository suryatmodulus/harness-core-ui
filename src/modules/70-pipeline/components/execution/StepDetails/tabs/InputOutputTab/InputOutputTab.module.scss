/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.io-tab {
  font-size: var(--font-size-small);
  margin: var(--spacing-xlarge);
  padding: var(--spacing-xsmall);

  &:not([data-empty='true']) {
    box-shadow: var(--card-shadow);
  }

  .io-row {
    display: grid;
    grid-template-columns: 1fr 1fr;
    line-height: 2;
    column-gap: var(--spacing-xsmall);

    &:not(:last-child) {
      border-bottom: 1px solid var(--grey-200);
    }

    &.header {
      font-weight: bold;
    }

    .key {
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
      padding: var(--spacing-xsmall) var(--spacing-xsmall) var(--spacing-xsmall) var(--spacing-xxlarge);

      &:hover {
        background-color: var(--primary-1);
      }
    }

    .value {
      color: var(--grey-400);
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
      padding: var(--spacing-xsmall);
      &:hover {
        background-color: var(--primary-1);
      }
    }
    .valueText {
      display: flex;
    }
  }

  .panel {
    padding: 0 var(--spacing-xsmall);

    &:not(:last-child) {
      border-bottom: 1px solid var(--grey-200);
    }

    .panelTitle {
      display: flex;
      column-gap: var(--spacing-small);
      cursor: pointer;
      line-height: 2;
      align-items: center;
      padding: var(--spacing-xsmall) var(--spacing-xsmall) 0 var(--spacing-xsmall);
      color: var(--grey-400);
    }

    &[data-status='close'] .panelTitle :global(.bp3-icon) {
      transform: rotate(180deg);
    }

    & + & {
      border-top: 1px solid var(--grey-300);
      margin-top: var(--spacing-xsmall);
    }
  }
}
