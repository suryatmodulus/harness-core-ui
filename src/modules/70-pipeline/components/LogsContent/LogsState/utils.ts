/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { escapeStringRegexp } from '@common/utils/StringUtils'

import type { State } from './types'

export function getRegexForSearch(str: string): RegExp {
  return new RegExp(escapeStringRegexp(str), 'gi')
}

export function getDefaultReducerState(props: Partial<Pick<State, 'selectedStage' | 'selectedStep'>> = {}): State {
  return {
    units: [],
    logKeys: [],
    selectedStage: props.selectedStage || '',
    selectedStep: props.selectedStep || '',
    dataMap: {},
    searchData: {
      text: '',
      currentIndex: 0,
      linesWithResults: []
    }
  }
}

export const logsCache = new Map<string, string>()
