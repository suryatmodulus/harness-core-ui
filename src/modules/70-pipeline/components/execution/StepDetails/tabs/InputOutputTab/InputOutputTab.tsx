/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { isPlainObject, toPairs, startCase, isEmpty, isNil } from 'lodash-es'
import { Collapse as BPCollapse, Icon } from '@blueprintjs/core'
import cx from 'classnames'

import { useStrings } from 'framework/strings'
import { CopyText } from '@common/components/CopyText/CopyText'
import { toVariableStr } from '@common/utils/StringUtils'

import css from './InputOutputTab.module.scss'

const blackListKeys = ['step', 'parallel']

function Collapse(props: React.PropsWithChildren<{ title: string }>): React.ReactElement {
  const [isOpen, setIsOpen] = React.useState(true)

  function toggle(): void {
    setIsOpen(status => !status)
  }

  return (
    <div className={css.panel} data-status={isOpen ? 'open' : 'close'}>
      <div className={css.panelTitle} onClick={toggle}>
        <Icon icon="chevron-up" />
        <span>{props.title}</span>
      </div>
      <BPCollapse isOpen={isOpen}>{props.children}</BPCollapse>
    </div>
  )
}

export interface InputOutputTabRowProps {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  data: Record<string, any>
  level: number
  prefix: string
}

export function InputOutputTabRow(props: InputOutputTabRowProps): React.ReactElement {
  return (
    <React.Fragment>
      {toPairs(props.data).map(([key, value]) => {
        if (key.startsWith('_') || isNil(value)) return null

        let newKey = `${props.prefix}.${key}`

        if (blackListKeys.includes(key.toLowerCase()) || key.toLowerCase().endsWith('definition')) {
          newKey = props.prefix
        }

        if (isPlainObject(value)) {
          if (isEmpty(value)) return null

          return (
            <Collapse key={key} title={startCase(key)}>
              <InputOutputTabRow prefix={newKey} data={value} level={props.level + 1} />
            </Collapse>
          )
        }

        if (Array.isArray(value)) {
          if (isEmpty(value)) return null

          if (value.every(e => typeof e === 'string')) {
            return (
              <div className={css.ioRow} key={key}>
                <div className={css.key}>
                  <CopyText textToCopy={toVariableStr(newKey)}>{key}</CopyText>
                </div>
                <div className={css.value}>
                  <CopyText textToCopy={value.join(', ')}>{value.join(', ')}</CopyText>
                </div>
              </div>
            )
          }

          return (
            <Collapse key={key} title={startCase(key)}>
              {value.map((item, index) => {
                return (
                  <InputOutputTabRow
                    key={`${newKey}[${index}]`}
                    prefix={`${newKey}[${index}]`}
                    data={item}
                    level={props.level + 1}
                  />
                )
              })}
            </Collapse>
          )
        }

        return (
          <div className={css.ioRow} key={key}>
            <div data-fqn={newKey} className={css.key}>
              <CopyText textToCopy={toVariableStr(newKey)}>{key}</CopyText>
            </div>
            <div className={css.value}>
              <CopyText textToCopy={value.toString()} className={css.valueText}>
                {value.toString()}
              </CopyText>
            </div>
          </div>
        )
      })}
    </React.Fragment>
  )
}

export interface InputOutputTabProps {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  data?: Record<string, any>
  mode: 'input' | 'output'
  baseFqn?: string
}

export function InputOutputTab(props: InputOutputTabProps): React.ReactElement {
  const { mode, baseFqn = '', data } = props
  const { getString } = useStrings()

  if (!data || isEmpty(data)) {
    return (
      <div className={css.ioTab} data-empty="true">
        {getString(
          mode === 'output' ? 'pipeline.execution.iotab.noOutputText' : 'pipeline.execution.iotab.noInputText'
        )}
      </div>
    )
  }

  return (
    <div className={css.ioTab}>
      <div className={cx(css.ioRow, css.header)}>
        <div>{getString(mode === 'input' ? 'inputName' : 'outputName')}</div>
        <div>{getString(mode === 'input' ? 'inputValue' : 'outputValue')}</div>
      </div>
      <InputOutputTabRow prefix={baseFqn} data={data} level={0} />
    </div>
  )
}
