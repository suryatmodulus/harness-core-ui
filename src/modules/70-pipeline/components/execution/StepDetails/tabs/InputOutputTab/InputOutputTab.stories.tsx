/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react'
import type { Meta, Story } from '@storybook/react'

import { TestWrapper } from '@common/utils/testUtils'
import { InputOutputTabProps, InputOutputTab } from './InputOutputTab'
import data from './__tests__/io-1.json'

export default {
  title: 'Pipelines / Execution / InputOutputTab ',
  component: InputOutputTab
} as Meta

type StoryProps = Pick<InputOutputTabProps, 'mode'> & {
  inputData: any
  outputData: any
}

export const Basic: Story<StoryProps> = args => (
  <TestWrapper>
    <InputOutputTab data={args.mode === 'input' ? args.inputData : args.outputData} mode={args.mode} />
  </TestWrapper>
)

Basic.args = {
  mode: 'input',
  inputData: [data.stepParameters],
  outputData: data.outcomes
}
