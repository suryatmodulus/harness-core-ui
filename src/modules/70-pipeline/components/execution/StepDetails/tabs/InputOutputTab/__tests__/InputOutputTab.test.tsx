/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'

import { TestWrapper } from '@common/utils/testUtils'
import { InputOutputTab } from '../InputOutputTab'
import data from './io-1.json'
describe('<ExecutionStepInputOutputTab /> tests', () => {
  test('output snapshot test', () => {
    const { container } = render(
      <TestWrapper>
        <InputOutputTab data={data.outcomes} mode="output" baseFqn="basefqn" />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test('input snapshot test', () => {
    const { container } = render(
      <TestWrapper>
        <InputOutputTab data={data.stepParameters} mode="input" baseFqn="basefqn" />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
