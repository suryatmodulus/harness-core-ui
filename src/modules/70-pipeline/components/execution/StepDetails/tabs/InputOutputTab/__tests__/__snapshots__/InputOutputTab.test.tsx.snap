// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExecutionStepInputOutputTab /> tests input snapshot test 1`] = `
<div>
  <div
    class="ioTab"
  >
    <div
      class="ioRow header"
    >
      <div>
        inputName
      </div>
      <div>
        inputValue
      </div>
    </div>
    <div
      class="panel"
      data-status="open"
    >
      <div
        class="panelTitle"
      >
        <span
          class="bp3-icon bp3-icon-chevron-up"
          icon="chevron-up"
        >
          <svg
            data-icon="chevron-up"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <desc>
              chevron-up
            </desc>
            <path
              d="M12.71 9.29l-4-4C8.53 5.11 8.28 5 8 5s-.53.11-.71.29l-4 4a1.003 1.003 0 001.42 1.42L8 7.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span>
          Pipeline Infrastructure
        </span>
      </div>
      <div
        class="bp3-collapse"
        style="height: auto; overflow-y: visible; transition: none;"
      >
        <div
          aria-hidden="false"
          class="bp3-collapse-body"
          style="transform: translateY(0); transition: none;"
        >
          <div
            class="panel"
            data-status="open"
          >
            <div
              class="panelTitle"
            >
              <span
                class="bp3-icon bp3-icon-chevron-up"
                icon="chevron-up"
              >
                <svg
                  data-icon="chevron-up"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    chevron-up
                  </desc>
                  <path
                    d="M12.71 9.29l-4-4C8.53 5.11 8.28 5 8 5s-.53.11-.71.29l-4 4a1.003 1.003 0 001.42 1.42L8 7.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <span>
                Infrastructure Definition
              </span>
            </div>
            <div
              class="bp3-collapse"
              style="height: auto; overflow-y: visible; transition: none;"
            >
              <div
                aria-hidden="false"
                class="bp3-collapse-body"
                style="transform: translateY(0); transition: none;"
              >
                <div
                  class="ioRow"
                >
                  <div
                    class="key"
                    data-fqn="basefqn.pipelineInfrastructure.type"
                  >
                    <div
                      class="main"
                    >
                      <div
                        class="text"
                      >
                        type
                      </div>
                    </div>
                  </div>
                  <div
                    class="value"
                  >
                    <div
                      class="main valueText"
                    >
                      <div
                        class="text"
                      >
                        KubernetesDirect
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="panel"
                  data-status="open"
                >
                  <div
                    class="panelTitle"
                  >
                    <span
                      class="bp3-icon bp3-icon-chevron-up"
                      icon="chevron-up"
                    >
                      <svg
                        data-icon="chevron-up"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                      >
                        <desc>
                          chevron-up
                        </desc>
                        <path
                          d="M12.71 9.29l-4-4C8.53 5.11 8.28 5 8 5s-.53.11-.71.29l-4 4a1.003 1.003 0 001.42 1.42L8 7.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                    <span>
                      Spec
                    </span>
                  </div>
                  <div
                    class="bp3-collapse"
                    style="height: auto; overflow-y: visible; transition: none;"
                  >
                    <div
                      aria-hidden="false"
                      class="bp3-collapse-body"
                      style="transform: translateY(0); transition: none;"
                    >
                      <div
                        class="ioRow"
                      >
                        <div
                          class="key"
                          data-fqn="basefqn.pipelineInfrastructure.spec.connectorRef"
                        >
                          <div
                            class="main"
                          >
                            <div
                              class="text"
                            >
                              connectorRef
                            </div>
                          </div>
                        </div>
                        <div
                          class="value"
                        >
                          <div
                            class="main valueText"
                          >
                            <div
                              class="text"
                            >
                              account.KubernetesConnectorForAutomationTestasd
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="ioRow"
                      >
                        <div
                          class="key"
                          data-fqn="basefqn.pipelineInfrastructure.spec.namespace"
                        >
                          <div
                            class="main"
                          >
                            <div
                              class="text"
                            >
                              namespace
                            </div>
                          </div>
                        </div>
                        <div
                          class="value"
                        >
                          <div
                            class="main valueText"
                          >
                            <div
                              class="text"
                            >
                              ng-testing
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="ioRow"
                      >
                        <div
                          class="key"
                          data-fqn="basefqn.pipelineInfrastructure.spec.releaseName"
                        >
                          <div
                            class="main"
                          >
                            <div
                              class="text"
                            >
                              releaseName
                            </div>
                          </div>
                        </div>
                        <div
                          class="value"
                        >
                          <div
                            class="main valueText"
                          >
                            <div
                              class="text"
                            >
                              release-pipeline-1
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="ioRow"
                      >
                        <div
                          class="key"
                          data-fqn="basefqn.pipelineInfrastructure.spec.releaseNumber"
                        >
                          <div
                            class="main"
                          >
                            <div
                              class="text"
                            >
                              releaseNumber
                            </div>
                          </div>
                        </div>
                        <div
                          class="value"
                        >
                          <div
                            class="main valueText"
                          >
                            <div
                              class="text"
                            >
                              6
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="ioRow"
                      >
                        <div
                          class="key"
                          data-fqn="basefqn.pipelineInfrastructure.spec.foo"
                        >
                          <div
                            class="main"
                          >
                            <div
                              class="text"
                            >
                              foo
                            </div>
                          </div>
                        </div>
                        <div
                          class="value"
                        >
                          <div
                            class="main valueText"
                          >
                            <div
                              class="text"
                            >
                              true
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="panel"
            data-status="open"
          >
            <div
              class="panelTitle"
            >
              <span
                class="bp3-icon bp3-icon-chevron-up"
                icon="chevron-up"
              >
                <svg
                  data-icon="chevron-up"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    chevron-up
                  </desc>
                  <path
                    d="M12.71 9.29l-4-4C8.53 5.11 8.28 5 8 5s-.53.11-.71.29l-4 4a1.003 1.003 0 001.42 1.42L8 7.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
              <span>
                Environment
              </span>
            </div>
            <div
              class="bp3-collapse"
              style="height: auto; overflow-y: visible; transition: none;"
            >
              <div
                aria-hidden="false"
                class="bp3-collapse-body"
                style="transform: translateY(0); transition: none;"
              >
                <div
                  class="ioRow"
                >
                  <div
                    class="key"
                    data-fqn="basefqn.pipelineInfrastructure.environment.name"
                  >
                    <div
                      class="main"
                    >
                      <div
                        class="text"
                      >
                        name
                      </div>
                    </div>
                  </div>
                  <div
                    class="value"
                  >
                    <div
                      class="main valueText"
                    >
                      <div
                        class="text"
                      >
                        env1
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ioRow"
                >
                  <div
                    class="key"
                    data-fqn="basefqn.pipelineInfrastructure.environment.identifier"
                  >
                    <div
                      class="main"
                    >
                      <div
                        class="text"
                      >
                        identifier
                      </div>
                    </div>
                  </div>
                  <div
                    class="value"
                  >
                    <div
                      class="main valueText"
                    >
                      <div
                        class="text"
                      >
                        env
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ioRow"
                >
                  <div
                    class="key"
                    data-fqn="basefqn.pipelineInfrastructure.environment.type"
                  >
                    <div
                      class="main"
                    >
                      <div
                        class="text"
                      >
                        type
                      </div>
                    </div>
                  </div>
                  <div
                    class="value"
                  >
                    <div
                      class="main valueText"
                    >
                      <div
                        class="text"
                      >
                        Production
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ExecutionStepInputOutputTab /> tests output snapshot test 1`] = `
<div>
  <div
    class="ioTab"
  >
    <div
      class="ioRow header"
    >
      <div>
        outputName
      </div>
      <div>
        outputValue
      </div>
    </div>
    <div
      class="panel"
      data-status="open"
    >
      <div
        class="panelTitle"
      >
        <span
          class="bp3-icon bp3-icon-chevron-up"
          icon="chevron-up"
        >
          <svg
            data-icon="chevron-up"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <desc>
              chevron-up
            </desc>
            <path
              d="M12.71 9.29l-4-4C8.53 5.11 8.28 5 8 5s-.53.11-.71.29l-4 4a1.003 1.003 0 001.42 1.42L8 7.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span>
          Output
        </span>
      </div>
      <div
        class="bp3-collapse"
        style="height: auto; overflow-y: visible; transition: none;"
      >
        <div
          aria-hidden="false"
          class="bp3-collapse-body"
          style="transform: translateY(0); transition: none;"
        >
          <div
            class="ioRow"
          >
            <div
              class="key"
              data-fqn="basefqn.output.type"
            >
              <div
                class="main"
              >
                <div
                  class="text"
                >
                  type
                </div>
              </div>
            </div>
            <div
              class="value"
            >
              <div
                class="main valueText"
              >
                <div
                  class="text"
                >
                  environmentOutcome
                </div>
              </div>
            </div>
          </div>
          <div
            class="ioRow"
          >
            <div
              class="key"
              data-fqn="basefqn.output.name"
            >
              <div
                class="main"
              >
                <div
                  class="text"
                >
                  name
                </div>
              </div>
            </div>
            <div
              class="value"
            >
              <div
                class="main valueText"
              >
                <div
                  class="text"
                >
                  env1
                </div>
              </div>
            </div>
          </div>
          <div
            class="ioRow"
          >
            <div
              class="key"
              data-fqn="basefqn.output.identifier"
            >
              <div
                class="main"
              >
                <div
                  class="text"
                >
                  identifier
                </div>
              </div>
            </div>
            <div
              class="value"
            >
              <div
                class="main valueText"
              >
                <div
                  class="text"
                >
                  env
                </div>
              </div>
            </div>
          </div>
          <div
            class="ioRow"
          >
            <div
              class="key"
              data-fqn="basefqn.output.environmentType"
            >
              <div
                class="main"
              >
                <div
                  class="text"
                >
                  environmentType
                </div>
              </div>
            </div>
            <div
              class="value"
            >
              <div
                class="main valueText"
              >
                <div
                  class="text"
                >
                  Production
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="panel"
      data-status="open"
    >
      <div
        class="panelTitle"
      >
        <span
          class="bp3-icon bp3-icon-chevron-up"
          icon="chevron-up"
        >
          <svg
            data-icon="chevron-up"
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <desc>
              chevron-up
            </desc>
            <path
              d="M12.71 9.29l-4-4C8.53 5.11 8.28 5 8 5s-.53.11-.71.29l-4 4a1.003 1.003 0 001.42 1.42L8 7.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span>
          Someotherkey
        </span>
      </div>
      <div
        class="bp3-collapse"
        style="height: auto; overflow-y: visible; transition: none;"
      >
        <div
          aria-hidden="false"
          class="bp3-collapse-body"
          style="transform: translateY(0); transition: none;"
        >
          <div
            class="ioRow"
          >
            <div
              class="key"
              data-fqn="basefqn.someotherkey.type"
            >
              <div
                class="main"
              >
                <div
                  class="text"
                >
                  type
                </div>
              </div>
            </div>
            <div
              class="value"
            >
              <div
                class="main valueText"
              >
                <div
                  class="text"
                >
                  KubernetesDirect
                </div>
              </div>
            </div>
          </div>
          <div
            class="ioRow"
          >
            <div
              class="key"
              data-fqn="basefqn.someotherkey.connectorRef"
            >
              <div
                class="main"
              >
                <div
                  class="text"
                >
                  connectorRef
                </div>
              </div>
            </div>
            <div
              class="value"
            >
              <div
                class="main valueText"
              >
                <div
                  class="text"
                >
                  account.KubernetesConnectorForAutomationTestasd
                </div>
              </div>
            </div>
          </div>
          <div
            class="ioRow"
          >
            <div
              class="key"
              data-fqn="basefqn.someotherkey.namespace"
            >
              <div
                class="main"
              >
                <div
                  class="text"
                >
                  namespace
                </div>
              </div>
            </div>
            <div
              class="value"
            >
              <div
                class="main valueText"
              >
                <div
                  class="text"
                >
                  ng-testing
                </div>
              </div>
            </div>
          </div>
          <div
            class="ioRow"
          >
            <div
              class="key"
              data-fqn="basefqn.someotherkey.releaseName"
            >
              <div
                class="main"
              >
                <div
                  class="text"
                >
                  releaseName
                </div>
              </div>
            </div>
            <div
              class="value"
            >
              <div
                class="main valueText"
              >
                <div
                  class="text"
                >
                  release-pipeline-1
                </div>
              </div>
            </div>
          </div>
          <div
            class="ioRow"
          >
            <div
              class="key"
              data-fqn="basefqn.someotherkey.kind"
            >
              <div
                class="main"
              >
                <div
                  class="text"
                >
                  kind
                </div>
              </div>
            </div>
            <div
              class="value"
            >
              <div
                class="main valueText"
              >
                <div
                  class="text"
                >
                  KubernetesDirect
                </div>
              </div>
            </div>
          </div>
          <div
            class="ioRow"
          >
            <div
              class="key"
            >
              <div
                class="main"
              >
                <div
                  class="text"
                >
                  path
                </div>
              </div>
            </div>
            <div
              class="value"
            >
              <div
                class="main"
              >
                <div
                  class="text"
                >
                  path1, path2
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
